src/mesh/BoundingBox.cpp

```diff
@@ -0,0 +1,171 @@
+#include "BoundingBox.hpp"
+
+namespace precice {
+namespace mesh {
+
+BoundingBox::BoundingBox(std::vector<double> bounds)
+{
+  //PRECICE_CHECK(bounds.size() == 4 || bounds.size() == 6, "Dimension of the bounding box should be 2 or 3.");
+  _bounds = bounds;
+  _dimensions = _bounds.size() / 2;
+}
+
+BoundingBox::BoundingBox(int dimension)
+:_dimensions(dimension)
+{
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
+
+BoundingBox::BoundingBox()
+{
+  _dimensions = 3;
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
```

Comment:
This one should not be needed.

Reply:
As I see `BoundingBoxMap` which is `std::map` requires a default constructor for the object. I tried to find a way but I could not find yet.

Reply:
@fsimonis What would be good practice here?

Reply:
The default constructor is only required if we use `std::map::operator[]`, as it default-constructs missing entries.
To fix that, we need to use `.at(key)` for lookup and `.emplace(key, value)` for insertion. ( Which makes the code more readable/expressive too. )

Reply:
Oh, nice then. I will modify the BoundingBoxMap insertions then. Thank you!